src/Controls/src/Core/HandlerImpl/NavigationPage/NavigationPage.Impl.cs

```diff
@@ -33,26 +33,10 @@ public partial class NavigationPage : INavigationView, IToolbarElement
 
 		Thickness IView.Margin => Thickness.Zero;
 
-		protected override Size MeasureOverride(double widthConstraint, double heightConstraint)
+		protected override void LayoutChildren(double x, double y, double width, double height)
 		{
-			if (Content is IView view)
-			{
-				_ = view.Measure(widthConstraint, heightConstraint);
-			}
-
-			return new Size(widthConstraint, heightConstraint);
-		}
-
-		protected override Size ArrangeOverride(Rectangle bounds)
```

Comment:
AFAICT this code was never getting called. Maybe this was just here for legacy shimmed navigation pages? This code was causing issues on iOS because it kept trying to measure the page as large as the NavigationPage. I realize the fix for this is to use `PageContainerArea` but I don't think we really need to use `PageContainerArea`. Each platform just sets the platform size of the pages and then from there everything seems to propagate up to the xplat code successfully

Reply:
I think you're correct that this was just for making the shimmed stuff work.